### ISSUE TITLE
Invisible album view after adding app shortcut commits

### ISSUE THREAD

**Issue by ArolWright** [adriancampos/Phonograph#2]
I have added the app shortcut commits to my Phonograph fork for personal use. However, after adding them, whenever you tap on an album or artist, it has no background, and letters look like they're free floating over the main activity. [Here's a screenshot](http://i.imgur.com/oG85yGc.png) of the album detail activity. This wasn't happening before adding the shortcuts.

Logcat only outputs the following:

D/ViewRootImpl[AlbumDetailActivity]: changeCanvasOpacity: opaque=false
W/org.jaudiotagger.audio:  /storage/40C0-B132/Music/Ed Sheeran - Eraser.mp3:Checking further because the ID3 Tag ends at 0x1043 but the mp3 audio doesnt start until 0x10e0

Edit: After locking the phone, it changed opaque=false to opaque=true, and it's now showing the album details correctly.

**Comment by adriancampos** [comment]
Were you able to reproduce this at will? I haven't seen it happen yet on my device.
Thanks for the report though, I'll keep an eye out.

**Comment by ArolWright** [comment]
Yep, it happens whenever I tap on an album or artist. It looks completely transparent, and after putting the app in background then going back it looks like it should. I have added all commits except for the "Autoformat" commit, which I've omitted because I was in a hurry. Maybe it has something to do?

I'll try and add that one when I have time. and I can record a video of the issue in question if it's necessary.

**Comment by adriancampos** [comment]
I'm experiencing this too, now.

It looks like [my use of the singleInstance launch mode](https://github.com/adriancampos/Phonograph/commit/aa90db02dff7d89aa90beb19e9c62af1e98f8ac7#diff-c8cc2dd42271f2bf39c6aa81eb6a5529R24) messed it up. I added that line so that the app shortcuts would work as intended when the app's already open in the background. Before that commit, the app would simply return to foreground.

Removing only that line yields a slightly better result, but:
- 1. If the app is started normally first (https://youtu.be/7eEmChLjv6A):
  - 1. A visual glitch occurs upon using the app shortcuts as Android brings the app to the foreground before it's killed
  - 2. The back stack contains all previous instances of the app.
- 2. If the app is started with an app shortcut (https://youtu.be/hsbZ1IkpsNs):
  - 1. The other app shortcuts don't work. They simply bring the app into foreground. 

Basically, the app shortcuts should only allow one instance of the app to be alive. Is there a way to launch an activity as singleinstance without using `android:launchMode="singleInstance"` in the manifest?

I'll remove that before submitting a pull request, but I would like to find a way to properly launch the app via app shortcuts. Do you have any ideas?
Thanks again for catching this! :)

**Comment by adriancampos** [comment]
It looks like adding the `Intent.FLAG_ACTIVITY_CLEAR_TOP` flag to the intent in [AppShortcutLauncherActivity](https://github.com/adriancampos/Phonograph/blob/feature/appshortcuts/app/src/main/java/com/kabouzeid/gramophone/appshortcuts/AppShortcutLauncherActivity.java#L86) fixes 1.b.

The visual glitch still occurs, and it has no effect when the app is started for the first time with an app shortcut.

**Comment by ArolWright** [comment]
Sorry for taking long to answer, I've been quite busy these days. How did you solve it?

**Comment by adriancampos** [comment]
That's quite alright! I've been pretty busy myself.
The issue occurred because I changed MainActivity's ```android:launchMode``` to singleInstance. To solve it, I reverted its launchmode back to the default, and solved the issue that I was having by adding the launch flag programmatically only when launched from an app shortcut (64520c6)
